### ISSUE TITLE
Add reverse_proxy transport

### ISSUE THREAD

**Issue by willnorris** [tailscale/caddy-tailscale#5]
We should implement a [reverse_proxy transport](https://caddyserver.com/docs/caddyfile/directives/reverse_proxy#transports) so that caddy can serve as a simple proxy to expose a tailscale host onto the public internet.

I would hope to enable a config sort of like:

```
mypublichost.example.com {
    reverse_proxy http://node.tails-scales.ts.net {
        transport tailscale
    }
}
```

This would require a tsnet Server that is used for making the outbound connection.  Currently, we setup that server only on a bind directive for listening, so we'd have to really think through how that should work.

Related public discussion: https://twitter.com/simonw/status/1579897624785670144  (of course, this is also doable with a separate tailscaled, but it'd be cool to support it in the caddy module)

**Comment by mholt** [comment]
OH, YES. I will pay money for this!

Wait, or I could just help implement this.

**Comment by willnorris** [comment]
Fixed in #13
